api/v1beta2/cloudstackmachine_types.go

```diff
@@ -83,6 +83,12 @@ type CloudStackMachineSpec struct {
 	// +optional
 	// +k8s:conversion-gen=false
 	FailureDomainName string `json:"failureDomainName,omitempty"`
+
+	// UncompressedUserData specifies whether the user data is gzip-compressed.
+	// cloud-init has built-in support for gzip-compressed user data, ignition does not
+	//
+	// +optional
+	UncompressedUserData *bool `json:"uncompressedUserData,omitempty"`
```

Comment:
Does this need to be a pointer? The zero value of a `bool` will be false and we could add a the default marker to be abundantly clear.

```
+kubebuilder:default=false
+optional
```

What do you think?

Reply:
No worries. Do you have a use-case where we'd take some different action as a result of it being `nil` vs `false`?

Reply:
I used this approach mainly because there is no default value for this setting and i was not sure what it would evaluate to. This is also heavily inspired by [cluster-api-provider-aws where they take a similar approach](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/4dda2debcf94a2f9d6cbe0b862bfcd6296178832/pkg/cloud/scope/machine.go#L216).

Reply:
The only time we really need a pointer is to differentiate between a user setting the value or not. If we don't need that, then `+kubebuilder:default=false` should be fine and would make whatever interprets this simpler because they don't need to perform nil pointer checks.

Currently, I don't see a need to differentiate set vs not set. Are you able to clarify?

Reply:
Hi @chrisdoherty4 i looked a bit further into the reasoning for the pointer. Apparently this is part of the Kubernetes API conventions / best practices (see [here](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required). 

Other references:

https://github.com/crossplane/crossplane/issues/741
https://github.com/kubernetes-sigs/kubebuilder/issues/2109

Reply:
That's good to know, thanks for looking into it.